components/centraldashboard/app/stackdriver_metrics_service_test.ts

```diff
@@ -266,6 +275,26 @@ describe('StackdriverMetricsService', () => {
           value: 0.12,
         }
       ]);
+
+      // Call again to validate that cluster-name is cached
+      await service.getPodCpuUtilization(Interval.Last5m);
+      expect(mockFetch).toHaveBeenCalledWith(
+          'http://metadata.google.internal/computeMetadata/v1/instance/attributes/cluster-name',
+          {headers: {'Metadata-Flavor': 'Google'}});
+      expect(mockFetch).toHaveBeenCalledTimes(1);
+    });
+
+    it('Filters without cluster name', async () => {
+      const response = {ok: false} as unknown as fetch.Response;
```

Comment:
Why do you need to double cast this object?

Reply:
Normally, TS compiler doesn't like that I am not including the other mandatory properties of the Response interface. I think because we aren't explicitly compiling the test sources we're OK so I have just listed it as fetch.Response to simplify.

Reply:
I see, thanks!